curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md

```diff
@@ -84,7 +84,35 @@ async (getUserInput) => {
 You can send a POST request to `/api/threads/{board}` with form data including `text` and `delete_password`. The saved database record will have at least the fields `_id`, `text`, `created_on`(date & time), `bumped_on`(date & time, starts same as `created_on`), `reported` (boolean), `delete_password`, & `replies` (array).
 
 ```js
-
+async (getUserInput) => {
+  const date = new Date();
+  const text = `fcc_test_${date}`;
+  const deletePassword = 'delete_me';
+  const data = { text, delete_password: deletePassword };
+  const url = getUserInput('url');
+  const res = await fetch(url + '/api/threads/fcc_test', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(data)
+  });
+  if (res.ok) {
+    const checkData = await fetch(url + '/api/threads/fcc_test');
```

Comment:
The test is making this request to check the data. This appears to be a route that the user has to implement. So if the user doesn't make that route correctly, this test would not pass. Is there any way we can make it so this doesn't rely on other user created routes? Any ideas are welcome. I suppose we would have to go mess with the testing part of the boilerplate and make our own route? We could maybe just add a note that it will not pass until the other route is correct.

Reply:
Most of the backend tests do this, though? Short of implementing it for Campers in the boilerplate (behind the scenes?) I cannot see another way around this.

Reply:
As shaun has said most of the other backend tests i looked into did the tests this way, if there is a better solution i'm happy to implement it.

Reply:
Alright sounds like a good plan to me, I'll complete the rest of the tests for the challenge this way for now anyway and if we need to change it down the line i can do.

Reply:
Thanks for all your hard work on this @jonathankerr.

There seem to be some hidden routes that we use in `fcctesting.js`, but they're used for checking headers and the test suite.

I think what you did here is fine. For the project to work, learners will have to implement dynamic routes and show any existing threads. Having a testing board and creating test threads with a timestamp seems like a good way to test for this user story. Might also be good in the sense that learners can see the threads our tests are creating.

Like @moT01 mentioned, worst case scenario we go back later and figure out how to do this behind the scenes on a route like `/_api/threads/fcc_test`.

Reply:
i am currently working on the project and as its part of the Information Security batch and we are dealing with passwords, i thought it would be a good idea to hash passwords before i put them into my database(using bcrypt). But the current test wont let my solution pass, as it checks if input password matches with database password